### ISSUE TITLE
Added item unavailable until re-enabling extension

### ISSUE THREAD

**Issue by Eagle3386** [fnxweb/urllink#15]
Adding a new item under moz-extension://db1d30dc-3f4f-43a1-8b7b-77d1b74edee4/preferences.html -> <kbd>Edit Menu</kbd> won't make it available within the context menu afterwards.

Clicking <kbd>Apply</kbd> or even restarting Firefox does _not_ resolve the issue either.
Only pressing <kbd>CTRL</kbd>+<kbd>SHIFT</kbd>+<kbd>A</kbd>, clicking <kbd>•••</kbd>, then <kbd>Disable</kbd>, then <kbd>•••</kbd> again and finally <kbd>Enable</kbd> let's the newly added item show up.

Is there some way to inform Firefox of changes to an addon's provided context menu entry's sub-entries?
Kinda dirty hack, i. e. "disable->re-enable", is either forbidden right away or probably will be in the future.

I'd happily assist in fixing this if I'm pointed in the right direction. :wink:

**Comment by fnxweb** [comment]
Yeah, I noticed another setting recently that didn't take immediately.

It did all work originally, so there must have been a change under the hood that's borked it. Suspect a threading race condition (it's all asychronous now).

Actually, now I type this, it's likely the [URL Link] message that's sent out internally when prefs are saved to the other bits of the extension so they know that things have changed. If that's now going AWOL it could explain this.
